crates/sui-graphql-client/src/lib.rs

```diff
@@ -548,29 +543,29 @@ impl Client {
 
     /// Get the balance of all the coins owned by address for the provided coin type.
     /// Coin type will default to `0x2::coin::Coin<0x2::sui::SUI>` if not provided.
-    pub async fn balance(
-        &self,
-        address: Address,
-        coin_type: Option<&str>,
-    ) -> Result<Option<u128>, Error> {
+    pub async fn balance(&self, address: Address, coin_type: Option<&str>) -> Result<Option<u128>> {
         let operation = BalanceQuery::build(BalanceArgs {
             address,
             coin_type: coin_type.map(|x| x.to_string()),
         });
         let response = self.run_query(&operation).await?;
 
         if let Some(errors) = response.errors {
-            return Err(Error::msg(format!("{:?}", errors)));
+            return Err(Error::graphql_error(errors));
         }
 
         let total_balance = response
             .data
             .map(|b| b.owner.and_then(|o| o.balance.map(|b| b.total_balance)))
-            .ok_or_else(|| Error::msg("No data in response"))?
+            .ok_or_else(|| {
+                Error::from_error(
+                    Kind::Query,
+                    "Expected response data from query but got None",
+                )
```

Comment:
Given this does repeat itself frequently it may be worth having a custom constructor for this

Reply:
Yep, better like that!